core/domain/stats_domain_test.py

```diff
@@ -547,6 +720,106 @@ def test_validate(self):
             'Invalid action type: %s' % 'InvalidActionType')):
             playthrough.validate()
 
+    def test_validate_exp_id(self):
+        playthrough = stats_domain.Playthrough(
+            'exp_id1', 1, 'EarlyQuit', {
+                'state_name': {
+                    'value': 'state_name1'
+                },
+                'time_spent_in_exp_in_msecs': {
+                    'value': 200
+                }
+            }, [stats_domain.LearnerAction.from_dict({
+                'action_type': 'ExplorationStart',
+                'action_customization_args': {
+                    'state_name': {
+                        'value': 'state_name1'
+                    }
+                },
+                'schema_version': 1
+            })])
+        playthrough.validate()
+
+        playthrough.exp_id = 0
+        with self.assertRaisesRegexp(utils.ValidationError, (
+            'Expected exp_id to be a string')):
+            playthrough.validate()
+
+    def test_validate_issue_type(self):
+        playthrough = stats_domain.Playthrough(
+            'exp_id1', 1, 'EarlyQuit', {
+                'state_name': {
+                    'value': 'state_name1'
+                },
+                'time_spent_in_exp_in_msecs': {
+                    'value': 200
+                }
+            }, [stats_domain.LearnerAction.from_dict({
+                'action_type': 'ExplorationStart',
+                'action_customization_args': {
+                    'state_name': {
+                        'value': 'state_name1'
+                    }
+                },
+                'schema_version': 1
+            })])
+        playthrough.validate()
+
+        playthrough.issue_type = 0
+        with self.assertRaisesRegexp(utils.ValidationError, (
+            'Expected issue_type to be a string')):
+            playthrough.validate()
+
+    def test_validate_actions(self):
+        playthrough = stats_domain.Playthrough(
+            'exp_id1', 1, 'EarlyQuit', {
+                'state_name': {
+                    'value': 'state_name1'
+                },
+                'time_spent_in_exp_in_msecs': {
+                    'value': 200
+                }
+            }, [stats_domain.LearnerAction.from_dict({
+                'action_type': 'ExplorationStart',
+                'action_customization_args': {
+                    'state_name': {
+                        'value': 'state_name1'
+                    }
+                },
+                'schema_version': 1
+            })])
+        playthrough.validate()
+
+        playthrough.actions = 0
+        with self.assertRaisesRegexp(utils.ValidationError, (
+            'Expected actions to be a list')):
+            playthrough.validate()
+
+    def test_validate_issue_customization_args(self):
```

Comment:
Please rename to: `test_validate_non_dict_issue_customization_args`

Reply:
done